hydromt_wflow/workflows/demand.py

```diff
@@ -0,0 +1,367 @@
+        transform=transform,
+        shape=(height, width),
+        crs=4326,
+    )
+
+    # Some info regarding the original ds
+    ds.raster.clip_bbox(ds_like.raster.bounds)
+
+    # Some extra info from ds_like
+
+    # Scale the polulation data based on global glob cells
+    popu_down = popu_scaled.raster.reproject_like(
+        dda,
+        method="sum",
+    )
+    popu_redist = popu_down.raster.reproject_like(
+        popu_scaled,
+        method="nearest",
+    )
+    popu_redist = popu_scaled / popu_redist
+
+    # Setup downscaled non irigation domestic data
+    non_iri_down = non_iri_scaled.raster.reproject_like(
+        dda,
+        method="sum",
+    )
+
+    # Loop through the domestic variables
+    for var in ["dom_gross", "dom_net"]:
+        attrs = non_iri_scaled[var].attrs
+        new = non_iri_down[var].raster.reproject_like(
+            non_iri_scaled,
+            method="nearest",
+        )
+        new = new * popu_redist
+        new = new.fillna(0.0)
+        non_iri_scaled[var] = new
+        non_iri_scaled[var] = non_iri_scaled[var].assign_attrs(attrs)
+
+    return non_iri_scaled, popu_scaled
+
+
+def allocate(
+    ds_like: xr.Dataset,
+    min_area: float | int,
+    admin_bounds: object,
+    basins: xr.Dataset,
+    rivers: xr.Dataset,
+):
+    """_summary_.
+
+    _extended_summary_
+
+    Parameters
+    ----------
+    ds_like : xr.Dataset
+        _description_
+    min_area : float | int
+        _description_
+    admin_bounds : object
+        _description_
+    basins : xr.Dataset
+        _description_
+    rivers : xr.Dataset
+        _description_
+    """
+    # Split based on admin bounds
+    sub_basins = basins.copy()
+    sub_basins["uid"] = range(len(sub_basins))
+
+    if admin_bounds is not None:
+        sub_basins = basins.overlay(
+            admin_bounds,
+            how="union",
+        )
+        sub_basins = sub_basins[~sub_basins["value"].isna()]
+
+        # Remove unneccessary stuff
+        cols = sub_basins.columns.drop(["value", "geometry", "admin_id"]).tolist()
+        sub_basins.drop(cols, axis=1, inplace=True)
+        # Use this uid to dissolve on later
+        sub_basins["uid"] = range(len(sub_basins))
+
+        # Dissolve cut pieces back
+        for _, row in sub_basins.iterrows():
+            if not str(row.admin_id).lower() == "nan":
+                continue
+            touched = sub_basins[sub_basins.touches(row.geometry)]
+            uid = touched[touched["value"] == row.value].uid.values[0]
+            sub_basins.loc[sub_basins["uid"] == row.uid, "uid"] = uid
+        sub_basins = sub_basins.dissolve("uid", sort=False, as_index=False)
+
+    # Set the contain flag per geom
+    sub_basins = sub_basins.apply(lambda row: touch_intersect(row, rivers), axis=1)
+    sub_basins["sqkm"] = sub_basins.geometry.to_crs(3857).area / 1000**2
+    _count = 0
+
+    # Create touched and not touched by rivers datasets
+    while True:
```

Comment:
We can have a talk, yes :)

Reply:
Update, changed parts to grid based functionality. Vector was indeed slow. Using a detailed admin map for the Rhine basin took ages.

Reply:
And why not the outflow_idxs method above instead of your long while loop? You have to trust the user to use an uparea threshold more than their river but it's already the case for the setup_floodplains method so we can actually do a check on this in the setup method first :)
Unless you would like to allow for exchanges between subbasins in terms of water allocation areas? I guess this would mainly be for small coastal basins as in your example above?

Then you can maybe decide to interpolate/extrapolate with nearest neighbor?